azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -181,12 +182,13 @@ class PipelineRootStage(PipelineStage):
     :type on_disconnected_handler: Function
     """
 
-    def __init__(self):
+    def __init__(self, pipeline_configurations=None):
```

Comment:
>pipeline_configurations=None [](start = 23, length = 28)

This shouldn't be optional, since other parts of the code directly access the pipeline_configuration. There's no case where we wouldn't have this.

Reply:
I think actually this should be optional at this point. Not every root pipeline stage has the need for options to be implemented right now, and it adds a large amount of test overhead to make this default.